src/Lighthouses.java

```diff
@@ -59,65 +60,94 @@ public void setBoard() {
     public void putShips( char[][] tab ) {
         boolean check = false;
         System.out.println("Podaj koordynaty kom�rki w kt�r� chcesz wstawi� statek:");
+        System.out.println("Wpisz 'q' aby zako�czy� gr�, 'z' aby zdj�� statek, 's' aby sprawdzi� plansz�");
+        char quit = key.next().charAt(0);
+        if (quit == 'z') {
+            removeShip(board);
+        }
+        else if (quit == 's') {
+            checkLighthouses(board);
+            putShips(board);
+        }
+        else if(quit=='q')
+        {
+            System.exit(0);
+        }
+
         int indexNumberRow;
         int indexNumberColumn;
-        char quit='q';
-        while ((quit=key.next().charAt(0))!='q') {
-            System.out.println("Wpisz 'q' aby zako�czy� gr�, lub 'z' aby zdj�� statek");
-            if(quit=='z')
-            {
-                removeShip(board);
-            }
-            try {
-                System.out.print("Rz�d:");
-                indexNumberRow = key.nextInt();
-                System.out.print("Kolumna:");
-                indexNumberColumn = key.nextInt();
-                int j;
-                for (int i = indexNumberRow - 1; i <= indexNumberRow + 1; i++) {
-                    for ( j = indexNumberColumn - 1; j <= indexNumberColumn + 1; j++) {
-                        /*System.out.print(tab[i][j] + " ");*/
-                        if (tab[i][j] == '_') {
-                            check = true;
-                        } else {
-                            System.out.println("Spr�buj jeszcze raz:");
-                            printBoard(board);
-                            check = false;
-                            putShips(board);
-                        }
+        try {
+            System.out.print("Rz�d:");
+            indexNumberRow = key.nextInt();
+            System.out.print("Kolumna:");
+            indexNumberColumn = key.nextInt();
+            int j;
+            for (int i = indexNumberRow - 1; i <= indexNumberRow + 1; i++) {
+                for (j = indexNumberColumn - 1; j <= indexNumberColumn + 1; j++) {
+                    /*System.out.print(tab[i][j] + " ");*/
+                    if (tab[i][j] == '_') {
+                        check = true;
+                    } else {
+                        System.out.println("Spr�buj jeszcze raz:");
+                        printBoard(board);
+                        check = false;
+                        putShips(board);
                     }
                 }
-                if (check==true) {
-                    tab[indexNumberRow][indexNumberColumn] = 'x';
-                }
+            }
+            if (check == true) {
+                tab[indexNumberRow][indexNumberColumn] = 'x';
                 printBoard(board);
-            } catch (ArrayIndexOutOfBoundsException e) {
-                System.out.println("Poza zakresem tablicy, spr�buj jeszcze raz");
```

Comment:
Jak Twoja mama (o ile nie zna programowania) zagra w lighthouse, i napiszesz Jej "poza zakresem tablicy", to będzie to pomocne, czy raczej utrudni ? :D 

Daj komunikat dla człowieka, nie dla programisty :) "Brak pola na planszy", "Podałeś niepoprawne współrzędne statku. Podaj jeszcze raz." ? Coś w tym stylu... :)